compiler/next/lib/Queries/Context.cpp

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/Queries/Context.h"
+
+#include "chpl/Queries/QueryImpl.h"
+
+#include <cstdlib>
+#include <cassert>
+
+namespace chpl {
+
+using namespace chpl::querydetail;
+
+Context::Context()
+  : uniqueStringsTable(), queryDB(), queryDeps(),
+    currentRevisionNumber(1), lastPrepareToGCRevisionNumber(0), gcCounter(1) {
+}
+
+owned<Context> Context::build() {
+  return toOwned(new Context());
+}
+
+Context::~Context() {
+  // free all of the unique'd strings
+  for (auto& item: this->uniqueStringsTable) {
+    const char* s = item.second;
+    free((void*)s);
+  }
```

Comment:
I would just say `free(item.second);` since the cast is unnecessary and since it's a one-liner the curly braces aren't needed.

Reply:
Actually I get an error (from `-fpermissive`) without the cast, so I will leave it this way.

Reply:
`free()` requires a pointer to non-const, so `const_cast` could be used.

Reply:
FWIW I have not been convinced of the value of these C++ styles of cast. Enabling "Grep for all of the casts" was a reason to do it IIRC but it is not an operation I would ever do. If we wish to discuss that further though we should probably do it outside of this PR review though.